src/application/application.css

```diff
@@ -40,12 +65,31 @@ body {
   bottom: 0;
   width: 100%;
   /* Set the fixed height of the footer here */
-  height: 4em;
+  height: 3.5em;
   background-color: #f5f5f5;
 }
-.form-inline > * {
-   margin:0 0.5em;
-}
-.right {
+.next-button {
   float: right;
+}
+
+@media screen and (max-width: 575px) {
+  /**
+   * When the form breaks into multiple rows,
+   * ditch the embedded scrolling panel behaviour
+   * and tidy up the form
+   */
+  .panel-container {
+    display: block;
+    height: auto;
+  }
+  .footer {
+    height: auto;
+    position: static;
+  }
+  body {
+    margin-bottom: auto;
```

Comment:
The way I've done this is that the body has a margin at the bottom just big enough for the footer so that the footer can sit at the bottom without blocking anything. This line resets the margin on the body when the footer is no longer floating at the bottom